### PR TITLE
Remove fqdn dependency for node name

### LINKED ISSUE #7071: Remove fqdn dependency for node name for GCE
Actually appending fqdn suffix already causes issue for non-resolvable node names based on @cjcullen today:

Today when NodeController "registers" nodes in SyncCloud by getting bare instance names (https://github.com/GoogleCloudPlatform/kubernetes/blob/9949936b5ce9d205328269ea304db4ab5dce5a67/pkg/cloudprovider/gce/gce.go#L101) from GCE, and recreating the fqdn and append it to instance names(https://github.com/GoogleCloudPlatform/kubernetes/blob/9949936b5ce9d205328269ea304db4ab5dce5a67/pkg/cloudprovider/gce/gce.go#L427).

### PR DESCRIPTION
**Pull request by dchen1107**
Based on @nak3's commit: Replace hostname -f with uname -n (PR #6466)

Fixed #7071 

Ran e2e tests, and all passed:

Ran 34 of 39 Specs in 869.489 seconds
SUCCESS! -- 34 Passed | 0 Failed | 1 Pending | 4 Skipped I0420 13:52:02.795357   13155 driver.go:117] All tests pass

cc/ @cjcullen @nak3

### CODE REVIEW

**Comment by zmerlynn** [comment]
Needs rebase, too. :)

**Comment by dchen1107** [comment]
Looks like with the newly change in cluster/validate-cluster.sh, I don't need commit: 71fc3bfd3b8b4b688caa5690f1006dba6e5dc162 anymore. Very nice. Will rebase and repush shortly. Thanks!

**Comment by dchen1107** [comment]
Done with rebase. Thanks!

**Comment by zmerlynn** [comment]
You should make sure it still works with the new changes, though. I think @cjcullen had some concerns that the Gol validation code wouldn't work with this method.

**Comment by cjcullen** [comment]
This breaks the /validate (GetComponentStatus) endpoint on the apiserver.  [getServersToValidate()](https://github.com/GoogleCloudPlatform/kubernetes/blob/e079e23e476b55e63d0749dc5f9a7310e9023028/pkg/master/master.go#L552) currently uses node names as resolvable URLs for health checks.

**Comment by dchen1107** [comment]
I just checked /validate, and it works:

# curl 127.0.0.1:8080/validate
[
  {
    "component": "controller-manager",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "scheduler",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "etcd-0",
    "health": "success",
    "msg": "{\"action\":\"get\",\"node\":{\"dir\":true,\"nodes\":[{\"key\":\"/registry\",\"dir\":true,\"modifiedIndex\":3,\"createdIndex\":3}]}}\n",
    "err": "nil"
  },
  {
    "component": "node-0",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "node-1",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  }

Ok, I will re-run e2e with post-rebase change now.

**Comment by dchen1107** [comment]
Ok run e2e test again, and everything passed:

Ran 34 of 39 Specs in 793.454 seconds
SUCCESS! -- 34 Passed | 0 Failed | 1 Pending | 4 Skipped I0420 16:18:37.049219    5929 driver.go:96] All tests pass

Also /validate looks  like to me:

```
dawnchen@e2e-test-dawnchen-master:~$ curl 127.0.0.1:8080/validate
[
  {
    "component": "node-1",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "controller-manager",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "scheduler",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  },
  {
    "component": "etcd-0",
    "health": "success",
    "msg": "{\"action\":\"get\",\"node\":{\"dir\":true,\"nodes\":[{\"key\":\"/registry\",\"dir\":true,\"modifiedIndex\":3,\"createdIndex\":3}]}}\n",
    "err": "nil"
  },
  {
    "component": "node-0",
    "health": "success",
    "msg": "ok",
    "err": "nil"
  }
```

**Comment by dchen1107** [comment]
btw @nak3 did sign with cla.

**Comment by zmerlynn** [comment]
I believe CLA-bot is just verifying that you have @nak3's permission to submit this (that's what the comment says), but https://github.com/GoogleCloudPlatform/kubernetes/pull/6466#issuecomment-93985051 is that permission.

File: pkg/util/node.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func GetHostname(hostnameOverride string) string {
 	if string(hostname) == "" {
 		// Note: We use exec here instead of os.Hostname() because we
 		// want the FQDN, and this is the easiest way to get it.
```
Comment (cjcullen) [inline pkg/util/node.go]: Should probably update this comment.

Is there a reason we can't use os.HostName() now? I'd be fine changing the comment to a "TODO: figure out if we can just use os.Hostname()"

Comment (cjcullen) [inline pkg/util/node.go]: It looks like uname -n and os.Hostname() both eventually pull from "/proc/sys/kernel/hostname". I'm not sure if that is true for all distros.

**Comment by cjcullen** [comment]
LGTM after updated comment.

/validate will still work, as long as bare hostnames can be resolved. Going forward, we shouldn't depend on node name being resolvable on the network, but that can be done separately from this PR.

File: pkg/util/node.go (hunk 1 above)
Comment (dchen1107) [inline pkg/util/node.go]: os.Hostname() retrieves the information from /proc/sys/kernel/hostname directly, but on some os dist, uname -n invokes uname syscall, not from that file.

**Comment by dchen1107** [comment]
Dropped that comment. PTAL?

File: pkg/util/node.go (hunk 1 above)
Comment (dchen1107) [inline pkg/util/node.go]: I think you are right on os.HostName(). uname.2.html says:

       Part of the utsname information is also accessible via
       /proc/sys/kernel/{ostype, hostname, osrelease, version, domainname}.

Comment (dchen1107) [inline pkg/util/node.go]: Can I change that separately including other docs fix?

**Comment by cjcullen** [comment]
LGTM.

**Pull request merged**